plugin/hover.py

```diff
@@ -106,22 +109,35 @@ def run_async() -> None:
             if not listener:
                 return
             if not only_diagnostics:
-                self.request_symbol_hover_async(listener, hover_point)
-            self._diagnostics_by_config, covering = listener.diagnostics_touching_point_async(
-                hover_point, userprefs().show_diagnostics_severity_level)
-            if self._diagnostics_by_config:
-                self.show_hover(listener, hover_point, only_diagnostics)
-            if not only_diagnostics and userprefs().show_code_actions_in_hover:
-                actions_manager.request_for_region_async(
-                    self.view, covering, self._diagnostics_by_config,
-                    functools.partial(self.handle_code_actions, listener, hover_point))
+                self.request_symbol_hover_async(listener, point_or_region)
+            if isinstance(point_or_region, int):
+                hover_point = point_or_region
+                self._diagnostics_by_config, covering = listener.diagnostics_touching_point_async(
+                    hover_point, userprefs().show_diagnostics_severity_level)
+                if self._diagnostics_by_config:
+                    self.show_hover(listener, hover_point, only_diagnostics)
+                if not only_diagnostics and userprefs().show_code_actions_in_hover:
+                    actions_manager.request_for_region_async(
+                        self.view, covering, self._diagnostics_by_config,
+                        functools.partial(self.handle_code_actions, listener, hover_point))
 
         sublime.set_timeout_async(run_async)
 
-    def request_symbol_hover_async(self, listener: AbstractViewListener, point: int) -> None:
+    def request_symbol_hover_async(self, listener: AbstractViewListener, point_or_region: PointOrRegion) -> None:
         hover_promises = []  # type: List[Promise[ResolvedHover]]
-        document_position = text_document_position_params(self.view, point)
+        point = point_or_region if isinstance(point_or_region, int) else point_or_region.begin()
+
         for session in listener.sessions_async('hoverProvider'):
+            capability = session.get_capability('experimental.rangeHoverProvider')
+            range_hover_provider = False
+            if isinstance(capability, bool):
+                range_hover_provider = capability
+
+            if isinstance(point_or_region, sublime.Region) and range_hover_provider:
+                document_position = text_document_range_params(self.view, point_or_region)
+            else:
+                document_position = text_document_position_params(self.view, point_or_region)
+
```

Comment:
Please don't add extra blank lines.
The style of the LSP code generally avoid blank lines within methods.

Reply:
Same in other places. Have a look at the complete diff here on GH.